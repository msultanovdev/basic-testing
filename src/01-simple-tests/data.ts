type dataType = {
  number1: number;
  number2: number;
  sum: number;
  diff: number;
  mult: number;
  part: number;
  exp: number;
};

export const calculations: dataType[] = [
  {
    number1: 5,
    number2: 3,
    sum: 8,
    diff: 2,
    mult: 15,
    part: 1.6666666666666667,
    exp: 125,
  },
  {
    number1: -4,
    number2: 7,
    sum: 3,
    diff: -11,
    mult: -28,
    part: -0.5714285714285714,
    exp: -16384,
  },
  {
    number1: 9,
    number2: 2,
    sum: 11,
    diff: 7,
    mult: 18,
    part: 4.5,
    exp: 81,
  },
  {
    number1: 123,
    number2: 456,
    sum: 579,
    diff: -333,
    mult: 56088,
    part: 0.26973684210526316,
    exp: Infinity,
  },
  {
    number1: -8,
    number2: -6,
    sum: -14,
    diff: -2,
    mult: 48,
    part: 1.3333333333333333,
    exp: 0.000003814697265625,
  },
];
