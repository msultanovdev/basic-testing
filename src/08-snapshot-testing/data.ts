type Movie = { id: number; name: string; rating: number };
export const moviesList1: Movie[] = [
  { id: 1, name: 'Interstellar', rating: 10 },
  { id: 2, name: 'The dark Knight', rating: 8 },
  { id: 3, name: 'Hobbit', rating: 6 },
];
export const moviesList2: Movie[] = [
  { id: 4, name: 'Love story', rating: 6.7 },
  { id: 5, name: 'Zeland', rating: 8 },
  { id: 6, name: 'Power Rangers', rating: 6 },
];
