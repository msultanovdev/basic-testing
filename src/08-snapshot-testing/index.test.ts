import { moviesList1, moviesList2 } from './data';
import { generateLinkedList } from './index';

describe('generateLinkedList', (): void => {
  const expected = {
    value: { id: 1, name: 'Interstellar', rating: 10 },
    next: {
      value: { id: 2, name: 'The dark Knight', rating: 8 },
      next: {
        value: { id: 3, name: 'Hobbit', rating: 6 },
        next: {
          value: null,
          next: null,
        },
      },
    },
  };
  test('should generate linked list from values 1', (): void => {
    expect(generateLinkedList(moviesList1)).toStrictEqual(expected);
  });
  test('should generate linked list from values 2', (): void => {
    expect(generateLinkedList(moviesList2)).toMatchSnapshot();
  });
});
