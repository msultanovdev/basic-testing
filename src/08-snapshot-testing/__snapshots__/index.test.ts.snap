// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generateLinkedList should generate linked list from values 2 1`] = `
{
  "next": {
    "next": {
      "next": {
        "next": null,
        "value": null,
      },
      "value": {
        "id": 6,
        "name": "Power Rangers",
        "rating": 6,
      },
    },
    "value": {
      "id": 5,
      "name": "Zeland",
      "rating": 8,
    },
  },
  "value": {
    "id": 4,
    "name": "Love story",
    "rating": 6.7,
  },
}
`;
